src/plugins/kibana_usage_collection/server/collectors/event_loop_delays/rollups/integration_tests/daily_rollups.test.ts

```diff
@@ -6,25 +6,37 @@
  * Side Public License, v 1.
  */
 
+import moment, { type MomentInput } from 'moment';
 import type { Logger, ISavedObjectsRepository, SavedObject } from '@kbn/core/server';
 import {
   type TestElasticsearchUtils,
   type TestKibanaUtils,
   createTestServers,
   createRootWithCorePlugins,
 } from '@kbn/core-test-helpers-kbn-server';
-import { rollDailyData } from '../daily';
+
 import { metricsServiceMock } from '@kbn/core/server/mocks';
 
 import {
   SAVED_OBJECTS_DAILY_TYPE,
   serializeSavedObjectId,
   EventLoopDelaysDaily,
 } from '../../saved_objects';
-import moment from 'moment';
+import { rollDailyData } from '../daily';
 
 const eventLoopDelaysMonitor = metricsServiceMock.createEventLoopDelaysMonitor();
 
+/*
+ * Mocking the constructor of moment, so we can control the time of the day.
+ * This is to avoid flaky tests when starting to run before midnight and ending the test after midnight
+ * because the logic might remove one extra document since we moved to the next day.
+ */
+jest.doMock('moment', () => {
+  const mockedMoment = (date?: MomentInput) => moment(date ?? '2023-07-04T10:00:00.000Z');
```

Comment:
I intentionally went 1 month in the past to prove the mock also applies the "prod" logic and not only the tests